src/index.js

```diff
@@ -63,6 +76,10 @@ class Signer {
     )
     params.set('X-Amz-SignedHeaders', this.signedHeaders)
 
+    if (contentLength && contentLength > 0) {
```

Comment:
```suggestion
    if (contentLength > 0) {
```

With above change this should no longer be needed